### PR TITLE
Add OpenStack as a valid platform

### PR DESCRIPTION
**Pull request by flaper87**
I'm adding support for OpenStack to the various operators, controllers, and providers that manage an openshift cluster.

I've started by pushing this patch as it seems to be the first step in the flow. I'm not sure if this PR should depend on others so I'd really appreciate some guidance in the process. :smile:

### CODE REVIEW

File: pkg/controller/template/render.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ const (
 //                    /master/units/kubelet.tmpl
 //                           /files/hostname.tmpl
 //       templates/aws/worker/units/kubelet-dropin.conf.tmpl
+//       templates/openstack/worker/units/kubelet-dropin.conf.tmpl
```
Comment (abhinavdahiya) [inline pkg/controller/template/render.go]: Same only example.

File: pkg/controller/template/test_data/templates/openstack/master/units/kubelet.service [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        --bootstrap-kubeconfig=/etc/kubernetes/kubeconfig \
+        --rotate-certificates \
+        --kubeconfig=/var/lib/kubelet/kubeconfig \
+        --container-runtime=remote \
+        --container-runtime-endpoint=/var/run/crio/crio.sock \
+        --allow-privileged \
+        --node-labels=node-role.kubernetes.io/master \
+        --minimum-container-ttl-duration=6m0s \
+        --client-ca-file=/etc/kubernetes/ca.crt \
+        --cloud-provider=openstack \
```
Comment (abhinavdahiya) [inline pkg/controller/template/test_data/templates/openstack/master/units/kubelet.service]: @flaper87 openstack is a valid cloud provider, right?

File: pkg/operator/assets/bindata.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Code generated by go-bindata.
+// Code generated by go-bindata. DO NOT EDIT.
```
Comment (abhinavdahiya) [inline pkg/operator/assets/bindata.go]: Can you drop these changes to bindata.go; they are not required.

File: pkg/controller/template/test_data/templates/openstack/master/units/kubelet.service (hunk 2 above)
Comment (flaper87) [reply pkg/controller/template/test_data/templates/openstack/master/units/kubelet.service]: it is!

File: pkg/operator/assets/bindata.go (hunk 3 above)
Comment (flaper87) [reply pkg/operator/assets/bindata.go]: yup, will do! Thanks

**Comment by abhinavdahiya** [comment]
/lgtm
I will keep an eye on installer repo to make sure we don't regress. :)

**Pull request merged**